### ISSUE TITLE
Why break 32-bit users in version 1.1?

### ISSUE THREAD

**Issue by jamesbraza** [contourpy/contourpy#255]
`contourpy` is a dependency of `matplotlib` on Windows.  As of `contourpy==1.1.0` when trying to `pip install` with a 32-bit version of Python 3.10:

```python
Collecting contourpy
  Using cached contourpy-1.1.0.tar.gz (13.4 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [24 lines of output]
      + meson setup C:\Users\user\AppData\Local\Temp\pip-install-65z9xnn_\contourpy_7c0cd46ed3714fbc8141c096a1b74e5c C:\Users\user\AppData\Local\Temp\pip-install-65z9xnn_\contourpy_7c0cd46ed3714fbc8141c096a1b74e5c\.mesonpy-4amrz8ye\build -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --vsenv --native-file=C:\Users\user\AppData\Local\Temp\pip-install-65z9xnn_\contourpy_7c0cd46ed3714fbc8141c096a1b74e5c\.mesonpy-4amrz8ye\build\meson-python-native-file.ini
      The Meson build system
      Version: 1.1.1
      Source dir: C:\Users\user\AppData\Local\Temp\pip-install-65z9xnn_\contourpy_7c0cd46ed3714fbc8141c096a1b74e5c
      Build dir: C:\Users\user\AppData\Local\Temp\pip-install-65z9xnn_\contourpy_7c0cd46ed3714fbc8141c096a1b74e5c\.mesonpy-4amrz8ye\build
      Build type: native build
      Project name: contourpy
      Project version: 1.1.0
      Activating VS 17.5.3
      C++ compiler for the host machine: cl (msvc 19.35.32216.1 "Microsoft (R) C/C++ Optimizing Compiler Version 19.35.32216.1 for x64")
      C++ linker for the host machine: link link 14.35.32216.1
      Host machine cpu family: x86_64
      Host machine cpu: x86_64
      Program python found: YES (C:\path\to\venv\Scripts\python.exe)
      Did not find pkg-config by name 'pkg-config'
      Found Pkg-config: NO
      pybind11-config found: YES (C:\Users\user\AppData\Local\Temp\pip-build-env-1ay0kgqi\overlay\Scripts\pybind11-config.EXE) 2.10.4
      Run-time dependency pybind11 found: YES 2.10.4
      Configuring _build_config.py using configuration
      Need python for x86_64, but found x86

      ..\..\src\meson.build:5:10: ERROR: Python dependency not found

      A full log can be found at C:\Users\user\AppData\Local\Temp\pip-install-65z9xnn_\contourpy_7c0cd46ed3714fbc8141c096a1b74e5c\.mesonpy-4amrz8ye\build\meson-logs\meson-log.txt
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
```

I can see from the [`1.1.0` release notes](https://github.com/contourpy/contourpy/releases/tag/v1.1.0):

> removes building of all 32-bit wheels and macOS universal2 wheels

Why remove support for 32-bit users?

I work in the manufacturing space, where many old computing systems require 32-bit programs, including Python.

**Comment by ianthomas23** [comment]
ContourPy 1.1.0 doesn't have 32-bit wheels but previous versions do, they are all still available and you can use one of them. Unfortunately the default behaviour of `pip install` is to try to compile the latest version from source rather than using a previous version which has a binary wheel, so you need to be explicit about what you want with your `pip` commands.

The simplest option is
```
pip install "contourpy<1.1"
```
but this isn't ideal as you have to remember version numbers. The recommended approach is
```
pip install --only-binary :all: contourpy
```

As to why, the entire Python scientific stack is dropping generation of 32-bit wheels e.g.

- NumPy numpy/numpy#23717 numpy/numpy#23838
- Matplotlib matplotlib/matplotlib#26125
- ContourPy #249

Strictly speaking this isn't dropping support for 32-bit, the whole stack works fine on 32-bit as of today, but it is dropping the building and distribution of compiled 32-bit wheels. If you want to or have to continue to use 32-bit, you can do any of these:

1. Stick with using packages built before now
2. Build your own packages
3. Offer time and/or effort to those projects to continue building 32-bit wheels

**Comment by jamesbraza** [comment]
Thanks for the comprehensive response!  Gotchu that everyone is dropping 32-bit builds, I can imagine it's becoming increasingly exotic.

For my use case, I don't directly use `contourpy`, it's a dependency of `matplotlib`.  So I made this issue after adding an indirect dependency to my requirements via `contourpy<1.1 ; sys_platform=='win32'`.

The binary-only install option is a nice idea, maybe I will switch to that.  Thank you again, going to close this out as resolved 👍

**Comment by eli-schwartz** [comment]
This isn't a matter of dropping support for 32-bit.as far as I can tell.

Building from source should work fine, except that meson detected a 64-bit C++ and a 32-bit Python, and errored out with a mismatch.

If you try running pip install from a 32-bit MSVC dev command prompt it should work fine.

**Comment by eli-schwartz** [comment]
No problem. :) as a meson developer I just wanted to clarify that there should be no meson-specific challenge to compiling 32-bit wheels. And of course I have an unusually good level of insight into reading meson error logs, so why not provide whatever analysis I can. :)

**Comment by ianthomas23** [comment]
Given that NumPy are intending to continue shipping win32 wheels in the upcoming 1.26.0 release, I have restated shipping ContourPy win32 wheels. They are included in today's release candidate [1.1.1rc1](https://github.com/contourpy/contourpy/releases/tag/v1.1.1rc1) which also supports Python 3.12. I will retroactively add win32 wheels to the previous (1.1.0) release so that there isn't a gap.
